connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -174,6 +176,14 @@ public ConnectorStateInfo connectorStatus(String connectorName) {
         String url = endpointForResource(String.format("connectors/%s/status", connectorName));
         try {
             return mapper.readerFor(ConnectorStateInfo.class).readValue(executeGet(url));
+        } catch (ConnectRestException e) {
+            //  the connector doesn't exist in the cluster yet
+            if (e.statusCode() == Response.Status.NOT_FOUND.getStatusCode()) {
+                log.warn("Could not find connector '{}' in cluster.", connectorName);
+                return null;
```

Comment:
Do all the places where this method is called handle null? There is no JavaDoc that explains/summarizes what will be returned, so maybe add that JavaDoc. The AbstractHerder never returns null and instead throws a NotFoundException when the named connector is not found.

Reply:
I removed this check, and documented the behavior. thanks!